RPGManager_development/test/unit_test/character_sheet_unit_tests/ArmorsUnitTests.java

```diff
@@ -0,0 +1,142 @@
+		CharacterSheet characterSheet = new CharacterSheet("CharacterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, malformedInput);
+	}
+
+	private void expectExceptionWithMessage(Class<? extends Exception> exceptionClass, String message) {
+		thrown.expect(exceptionClass);
+		thrown.expectMessage(message);
+	}
+
+	@Test
+	public void testSetArmorTypeLightPlate() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.LIGHT_PLATE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_PLATE, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_MAIL, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForRogue() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.ROGUE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.LIGHT_LEATHER, actualValue);
+	}
+
+	@Test
+	public void testGetDefaultArmorForMage() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.MAGE);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.NONE, actualValue);
+	}
+
+	@Test
+	public void testSetArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.HEAVY_MAIL);
+		Armors actualValue = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(Armors.HEAVY_MAIL, actualValue);
+	}
+
+	@Test
+	public void testGetDexterityArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.ARMOR_TYPE, Armors.HEAVY_MAIL);
+		int actualValue = characterSheet.getData(Fields.DEXTERITY_VALUE);
+		assertEquals(-3, actualValue + Armors.class.cast(characterSheet.getData(Fields.ARMOR_TYPE)).getArmorPenalty());
+	}
+
+	@Test
+	public void testGetCostArmorForWarrior() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.WARRIOR);
+		Armors armor = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(50, armor.getCost());
+	}
+
+	@Test
+	public void testGetCostArmorForMage() {
+		CharacterSheet characterSheet = new CharacterSheet("characterSheet");
+		characterSheet.setData(Fields.BASECLASS, BaseClasses.MAGE);
+		Armors armor = characterSheet.getData(Fields.ARMOR_TYPE);
+		assertEquals(0, armor.getCost());
+	}
+
+	@Test
+	public void testGetDefaultDefenseAnderSurvivor() {

```

Comment:
"testGetDefaultDefenseAnderSurvivor" ez félrevezető. Semmi köze a Defense-hez (ami ugye dexből jön) lehetne "testGetDefaultArmorRating_HeavyPlate" nem beszélve arról hogy a többi számozott tesztet is hasonló logika alapján el tudod nevezni, nem fogom egyenként odaírni, de azokat a számozottakat nevezd el normálisan.

Egyébként erre egy parametrizált teszt lenne a legízlésesebb ahol nem csak szúrópróbaszerűen nézel meg egy-két értéket hanem az összes enumod default armor ratingjét végignyálazod. Pl. a heavy_leather-t nem látom ezek között de a light_mail-t sem.
Egyébként meg ehhez állítgatni is felesleges a karakterlapot. Végigtesztelheted az Armor.getArmorRatinggel....